bundles/org.openhab.binding.enocean/src/main/java/org/openhab/binding/enocean/internal/EnOceanHandlerFactory.java

```diff
@@ -36,66 +38,76 @@
             .collect(Collectors.toSet());
 
     private Map<ThingUID, ServiceRegistration<?>> discoveryServiceRegs = new HashMap<>();
 
-    @Reference
-    SerialPortManager serialPortManager;
-
-    @Reference
-    ItemChannelLinkRegistry itemChannelLinkRegistry;
+    private final SerialPortManager serialPortManager;
+    private final ThingManager thingManager;
+    private final ItemChannelLinkRegistry itemChannelLinkRegistry;
 
-    @Reference
-    ThingManager thingManager;
+    @Activate
+    public EnOceanHandlerFactory(final @Reference SerialPortManager serialPortManager,
+            final @Reference ThingManager thingManager,
+            final @Reference ItemChannelLinkRegistry itemChannelLinkRegistry) {
+        // Obtain references to thes service using an OSGi reference
+        this.serialPortManager = serialPortManager;
+        this.thingManager = thingManager;
+        this.itemChannelLinkRegistry = itemChannelLinkRegistry;
+    }
 
     @Override
     public boolean supportsThingType(ThingTypeUID thingTypeUID) {
         return SUPPORTED_THING_TYPES_UIDS.contains(thingTypeUID);
     }
 
     @Override
-    protected ThingHandler createHandler(Thing thing) {
+    protected @Nullable ThingHandler createHandler(Thing thing) {
         ThingTypeUID thingTypeUID = thing.getThingTypeUID();
-
+        ItemChannelLinkRegistry localItemChannelLinkRegistry = itemChannelLinkRegistry;
         if (EnOceanBridgeHandler.SUPPORTED_THING_TYPES.contains(thingTypeUID)) {
             EnOceanBridgeHandler bridgeHandler = new EnOceanBridgeHandler((Bridge) thing, serialPortManager);
             registerDeviceDiscoveryService(bridgeHandler);
             return bridgeHandler;
         } else if (EnOceanBaseActuatorHandler.SUPPORTED_THING_TYPES.contains(thingTypeUID)) {
-            return new EnOceanBaseActuatorHandler(thing, itemChannelLinkRegistry);
+            return new EnOceanBaseActuatorHandler(thing, localItemChannelLinkRegistry);
         } else if (EnOceanBaseSensorHandler.SUPPORTED_THING_TYPES.contains(thingTypeUID)) {
-            return new EnOceanBaseSensorHandler(thing, itemChannelLinkRegistry);
+            return new EnOceanBaseSensorHandler(thing, localItemChannelLinkRegistry);
         } else if (EnOceanClassicDeviceHandler.SUPPORTED_THING_TYPES.contains(thingTypeUID)) {
-            return new EnOceanClassicDeviceHandler(thing, itemChannelLinkRegistry);
+            return new EnOceanClassicDeviceHandler(thing, localItemChannelLinkRegistry);
         }
 
         return null;
     }
 
     @Override
     protected void removeHandler(ThingHandler thingHandler) {
-        if (this.discoveryServiceRegs != null) {
-            ServiceRegistration<?> serviceReg = this.discoveryServiceRegs.get(thingHandler.getThing().getUID());
-            if (serviceReg != null) {
-                serviceReg.unregister();
-                discoveryServiceRegs.remove(thingHandler.getThing().getUID());
-            }
+        @Nullable
```

Comment:
Added this to prevent the info message:
`Unsafe interpretation of method return type as '@NonNull' based on the receiver type '@NonNull Map<@NonNull ThingUID,@NonNull ServiceRegistration<?>>'. Type 'Map<K,V>' doesn't seem to be designed with null type annotations in mind`

Agree that it not needed. removed it.